AzureData/Source/Resources/ValidationError.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  ValidationError.swift
+//  AzureData
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+
+public enum ValidationError: Error {
```

Comment:
Yes. Here https://github.com/Azure/Azure.iOS/pull/59/files#diff-859410b5db30a90139edf43928aa0282R1045 and here https://github.com/Azure/Azure.iOS/pull/59/files#diff-859410b5db30a90139edf43928aa0282R1168

Reply:
@ftchirou  Can we incorporate it into DocumentClientError instead of having a custom type? Maybe keep the Dictionary extensions to validate, then return a new `DocumentClientError` `kind`?  I consolidated some other error types because I was worried that the user won't know to cast the error to the different types.
What are your thoughts?  I'm open to discuss if you feel strongly that this should be a unique type.

Reply:
Yes. I think this should be a unique type just for (client side) request validation purposes. We can have multiple validation errors per request (along with meaningful error messages). It's true that the user may not know the different types to cast to. Following your suggestion, how about having an additional kind to `DocumentClientError`, `case validationError(_: ValidationError)`? With this, it's obvious to the user that an error can be a `ValidationError`.

Reply:
I like that.  What do you think about renaming it to something more specific?  `ValidationError` is pretty generic, i.e. _what_ is it validating?

Reply:
I like `ResourceRequestError`